tests/integrationv2/test_client_authentication.py

```diff
@@ -3,87 +3,99 @@
+@pytest.mark.parametrize("certificate", ALL_CERTS, ids=get_parameter_name)
+def test_client_auth_with_s2n_server(managed_process, cipher, provider, curve, protocol, certificate):
     host = "localhost"
     port = next(available_ports)
 
+    # NOTE: Client Auth is failing for ECDSA client certs
+    if 'ecdsa' in certificate.cert:
+        pytest.skip("Skipping known failure, ECDSA client auth certificate")
+
     random_bytes = data_bytes(64)
     client_options = ProviderOptions(
-        mode="client",
+        mode=Provider.ClientMode,
         host="localhost",
         port=port,
         cipher=cipher,
         curve=curve,
         data_to_send=random_bytes,
         use_client_auth=True,
-        client_key_file='../pems/rsa_1024_sha256_client_key.pem',
-        client_certificate_file='../pems/rsa_1024_sha256_client_cert.pem',
+        client_key_file=certificate.key,
+        client_certificate_file=certificate.cert,
         insecure=False,
-        tls13=tls13)
+        protocol=protocol)
 
     server_options = copy.copy(client_options)
     server_options.data_to_send = None
-    server_options.mode = "server"
+    server_options.mode = Provider.ServerMode
     server_options.key = "../pems/rsa_2048_sha256_wildcard_key.pem"
     server_options.cert = "../pems/rsa_2048_sha256_wildcard_cert.pem"
 
     # Passing the type of client and server as a parameter will
     # allow us to use a fixture to enumerate all possibilities.
     server = managed_process(S2N, server_options, timeout=5)
-    client = managed_process(OpenSSL, client_options, timeout=5)
+    client = managed_process(provider, client_options, timeout=5)
 
     # The client should connect and return without error
     for results in client.get_results():
         assert results.exception is None
         assert results.exit_code == 0
 
+    expected_version = get_expected_s2n_version(protocol, provider)
+
     # S2N should indicate the procotol version in a successful connection.
     for results in server.get_results():
         assert results.exception is None
         assert results.exit_code == 0
-        if tls13:
-            assert b"Actual protocol version: 34" in results.stdout
-        else:
-            assert b"Actual protocol version: 33" in results.stdout
+        assert bytes("Actual protocol version: {}".format(expected_version).encode('utf-8')) in results.stdout
         assert random_bytes in results.stdout
 
 
-@pytest.mark.parametrize("cipher", CIPHERSUITES)
-@pytest.mark.parametrize("curve", CURVES)
-@pytest.mark.parametrize("tls13", TLS13)
-def test_client_auth_with_s2n_server_using_nonmatching_certs(managed_process, cipher, curve, tls13):
+@pytest.mark.uncollect_if(func=invalid_test_parameters)
+@pytest.mark.parametrize("cipher", [cipher for cipher in ALL_TEST_CIPHERS if 'ECDSA' not in cipher.name], ids=get_parameter_name)
+@pytest.mark.parametrize("provider", [OpenSSL])
+@pytest.mark.parametrize("curve", ALL_CURVES)
```

Comment:
Since Curve x25519 is not supported in libcrypto versions 1.0.2 or lower, we need to make sure that ALL_CURVES doesnot contain x25519 for these libcrypto versions.

Reply:
As above. The way we can address this is to de-select a test if the curve parameter is not supported by the test parameter.